test/k8sT/fqdn.go

```diff
@@ -168,6 +168,9 @@ var _ = Describe("K8sFQDNTest", func() {
 		ExpectAllPodsTerminated(kubectl)
 		ExpectCiliumReady(kubectl)
 
+		err = kubectl.CiliumEndpointWaitReady()
```

Comment:
When exactly is this needed? Is it just to avoid the flake on restart until we have figured out the reason why we still see some drops on restarts?

Reply:
Yes, my main point is to reduce the flake, my concern here: 

- There is a situation where a dns request has been used,  endpoint has the policy and Cilium is ready, but endpoint has a different port for DNS-proxy so DNS-proxy is not ready at all. That it's why we wait for the endpoint before the `connectivityReady()`

- There is this situation, where Cilium is healhty and endpoint can't connect to the IP. That it's weird because the IP is restored from endpoint model and the JSON is saved on each dnsHistory update.  

https://github.com/cilium/cilium/blob/6db1513578a0b0c6d72c91f2db452e680cbd36af/daemon/fqdn.go#L181-L186

It's a bit weird, because the IP should be in the endpoint config, and on restore the IP should be also whitelisted. IP is allowed when Cilium is down, so the lxc_config.h has been written correctly.  Hopefully this commit will help us if with the endpoints ready this also happens and get to the next steps.

Reply:
Resolving, @eloycoto added the comment giving appropriate context.